src/extension.ts

```diff
@@ -124,6 +124,8 @@ function tryToForceLanguageId(document: TextDocument, fileName: string, document
   });
 }
 
+const APP_PROPERTIES_PATTERN = /^application(?:-[^.]+)\.properties$/;
```

Comment:
On the quarkus-ls side, I used `[A-Za-z]` to match the profile name; figure out if non letter characters are supported by Quarkus.